tests/Controllers/TaskControllerTest.php

```diff
@@ -0,0 +1,160 @@
+        $this->withHeaders(['X-Requested-With' => 'XMLHttpRequest']);
+    }
+
+    private function mockRepository()
+    {
+        $this->taskRepository = \Mockery::mock(TaskRepository::class);
+        app()->instance(TaskRepository::class, $this->taskRepository);
+    }
+
+    public function tearDown(): void
+    {
+        parent::tearDown();
+        \Mockery::close();
+    }
+
+    /** @test */
+    public function test_can_retrieve_task()
+    {
+        /** @var Task $task */
+        $task = factory(Task::class)->create();
+
+        $user = factory(User::class)->create();
+        $task->taskAssignee()->sync([$user->id]);
+
+        $tag = factory(Tag::class)->create();
+        $task->tags()->sync([$tag->id]);
+
+        $response = $this->getJson('tasks/'.$task->id.'/edit');
+
+        $this->assertSuccessDataResponse($response, $task->toArray(), 'Task retrieved successfully.');
+    }
+
+    /** @test */
+    public function test_can_delete_task()
+    {
+        /** @var Task $task */
+        $task = factory(Task::class)->create();
+
+        $response = $this->deleteJson('tasks/'.$task->id);
+
+        $this->assertSuccessMessageResponse($response, 'Task deleted successfully.');
+
+        $response = $this->getJson('tasks/'.$task->id.'/edit');
+
+        $response->assertStatus(404);
+        $response->assertJson([
+            'success' => false,
+            'message' => 'Task not found.',
+        ]);
+    }
+
+    /** @test */
+    public function test_can_not_delete_task_when_task_has_time_entries()
+    {
+        /** @var TimeEntry $timeEntry */
+        $timeEntry = factory(TimeEntry::class)->create();
+
+        $response = $this->deleteJson('tasks/'.$timeEntry->task_id);
+
+        $response->assertJson([
+            'success' => false,
+            'message' => 'Task has one or more time entries.',
+        ]);
+    }
+
+    /** @test */
+    public function test_can_update_status()
+    {
+        $this->mockRepository();
+
+        /** @var Task $task */
+        $task = factory(Task::class)->create();
+
+        $this->taskRepository->shouldReceive('updateStatus')
+            ->once()
+            ->with($task->id);
+
+        $response = $this->postJson("tasks/$task->id/update-status");
+        $response->assertJson([
+            'success' => true,
+            'message' => 'Task status Update successfully.',
+        ]);
+    }
+
+    /** @test */
+    public function test_can_get_task_details()
+    {
+        $this->mockRepository();
+
+        /** @var Task $task */
+        $task = factory(Task::class)->create();
+
+        $this->taskRepository->shouldReceive('getTaskDetails')
+            ->once()
+            ->with($task->id);
+
+        $response = $this->getJson("task-details/$task->id");
+        $response->assertJson([
+            'success' => true,
+            'message' => 'Task retrieved successfully.',
+        ]);
+    }
+
+    /** @test */
+    public function test_can_get_my_tasks()
```

Comment:
don't use test case name as per function name, its should be as per whats going on into it.

`test_can_get_task_of_logged_in_user_for_given_project`